### PR TITLE
subclass support

### PR DESCRIPTION
**Pull request by messense**
Ref #15

### CODE REVIEW

File: pyo3cls/src/py_class.rs [hunk 1]
```diff
@@ -440,6 +440,10 @@ fn parse_attribute(attr: String) -> (HashMap<&'static str, syn::Ident>,
                         flags.push(syn::Ident::from("_pyo3::typeob::PY_TYPE_FLAG_WEAKREF"));
                         continue
                     }
+                    "basetype" => {
```
Comment (messense) [inline pyo3cls/src/py_class.rs]: I am sure this is a good name. Feels odd.

Comment (fafhrd91) [inline pyo3cls/src/py_class.rs]: agree. maybe "subclass"?

File: guide/src/class.md [hunk 2]
```diff
@@ -61,6 +61,7 @@ python object that can be collector `PyGCProtocol` trait has to be implemented.
 * `weakref` - adds support for python weak references
 * `base=xxx.YYY` - use custom base class. It is not possible to call constructor
 of base class at the moment. `xxx.YYY`, `xxx` - module name, `YYY` class name.
+* `basetype` - adds subclass support
```
Comment (fafhrd91) [inline guide/src/class.md]: maybe something like "python classes can inherit from this class", otherwise it is not clear

**Comment by messense** [comment]
Seem that some test cases are flaky.

**Comment by fafhrd91** [comment]
Yes, ref count related tests are flaky because Rust runs multiple tests in parallel. I will fix later

**Pull request merged**